projects/netgrif-components-core/src/lib/configuration/configuration.service.ts

```diff
@@ -246,6 +259,30 @@ export abstract class ConfigurationService {
         return config.providers.auth.address + config.providers.auth.endpoints[endpointKey];
     }
 
+    /**
+     * Loads the configuration for the application by fetching data from the resource provider
+     * and initializes the application settings based on the fetched configuration.
+     *
+     * @return {Promise<any>} A promise that resolves with the fetched configuration or returns null
+     * if the configuration resolution is not enabled.
+     */
+    public loadConfiguration(): Promise<any> {
```

Comment:
As far as I know, `APP_INITIALIZER` factory method can return `Observable`, so you don't have to recast it to `Promise`.